docs/design/baremetal/networking-infrastructure.md

```diff
@@ -70,7 +70,101 @@ DNS requirements internal to the cluster as possible, leaving only a
 small amount of public DNS configuration to be implemented by the user
 before starting the installation process.
 
-TODO - explain how this works ...
+In a `baremetal` environment, we do not know the IP addresses of all hosts in
+advance.  Those will come from an organization’s DHCP server.  Further, we can
+not rely on being able to program an organization’s DNS infrastructure in all
+cases.  We address these challenges in two ways:
 
-Explain that $cluster_name and $base_domain are supplied via
-install-config.yaml
+1. Self host some DNS infrastructure to provide DNS resolution for records only
+   needed internal to the cluster.
+2. Make use of mDNS (Multicast DNS) to dynamically discover the addresses of
+   hosts that we must resolve records for.
+
+### mdns-publisher
+
+https://github.com/openshift/mdns-publisher
+
+The `mdns-publisher` is the component that runs on each host to make itself
+discoverable by other hosts in the cluster.  It will be run by the
```

Comment:
On control plane machines, it advertises using etcd-NN and master-NN names, but on worker nodes only the worker-NN name is published

Reply:
Thanks, I've added this, including a note that the master-NN and worker-NN names may no longer be necessary AFAICT ...